torch_xla/csrc/ops/device_data.h

```diff
@@ -10,10 +10,18 @@ class DeviceData : public XlaNode {
  public:
   DeviceData(std::shared_ptr<torch::lazy::BackendData> data);
 
+  DeviceData(std::shared_ptr<torch::lazy::BackendData> data,
+             torch::lazy::OpList ops, xla::Shape xla_shape,
+             xla::OpSharding sharding);
+
   std::string ToString() const override;
 
+  torch::lazy::NodePtr Clone() const;
+
   torch::lazy::NodePtr Clone(torch::lazy::OpList operands) const override;
```

Comment:
Hmmm, this Clone method is only used in https://github.com/pytorch/xla/blob/master/torch_xla/csrc/ir_util.h#L15, and those two Util::Clone are not used anywhere...

I guess we can actually remove it. But let's do it in a separate patch. Could you leave a TODO in XlaNode::Clone.

Reply:
Added a TODO https://github.com/pytorch/xla/issues/4568